### PR TITLE
🤬 AML prepare

### PR DESCRIPTION
**Pull request by nikokaoja**
### Fixed
- Edge creation
### Added
- `.prepare.instances.aml()`

### CODE REVIEW

File: cognite/neat/_session/_prepare.py [hunk 1]
```diff
@@ -101,6 +101,40 @@ def dexpi(self) -> None:
...
 
+    def aml(self) -> None:
+        """Prepares extracted AutomationML graph for further usage in CDF
+
+        This method bundles several graph transformers which:
+        - attach values of attributes to nodes
+        - create associations between nodes
+        - remove unused generic attributes
+        - remove associations between nodes that do not exist in the extracted graph
+        - remove edges to nodes that do not exist in the extracted graph
```
Comment (nikokaoja) [inline cognite/neat/_session/_prepare.py:112]:
```suggestion
        - attach values of attributes to nodes
        - remove unused attributes
        - remove edges to nodes that do not exist in the extracted graph
```

File: tests/data/car.py [hunk 2]
```diff
@@ -24,15 +24,15 @@
...
-        (_neat["Car1"], _neat["Car.Manufacturer"], _neat["Toyota"]),
+        (_neat["Car1"], _neat["make"], _neat["Toyota"]),
         (_neat["Car1"], _neat["year"], Literal(2020)),
         (_neat["Car1"], _neat["color"], _neat["Blue"]),
         (_neat["Ford"], RDF.type, _neat["Manufacturer"]),
         (_neat["Ford"], _neat["name"], Literal("Ford")),
         (_neat["Red"], RDF.type, _neat["Color"]),
         (_neat["Red"], _neat["name"], Literal("red")),
         (_neat["Car2"], RDF.type, _neat["Car"]),
-        (_neat["Car2"], _neat["Car.Manufacturer"], _neat["Ford"]),
```
Comment (nikokaoja) [inline tests/data/car.py:35]: this was wrong, as actual property that holds edges is `make`, `Car.Manufacturer` is edge type

File: tests/tests_integration/test_session/test_graph_flow.py [hunk 3]
```diff
@@ -99,10 +103,53 @@ def test_dexpi_to_dms(self, cognite_client: CogniteClient, data_regression: Data
...
+        rules_str = neat.to.yaml(format="neat")
+
+        rules_dict = yaml.safe_load(rules_str)
+        data_regression.check(
+            {
+                "rules": rules_dict,
+                "instances": [],
+            }
+        )
+
```
Comment (nikokaoja) [inline tests/tests_integration/test_session/test_graph_flow.py:123]: we did not test if and how many edges are we creating, we only look at raw no of instances and missed the fact that we were not producing edges for some time

File: cognite/neat/_graph/loaders/_rdf2dms.py [hunk 4]
```diff
@@ -197,7 +197,7 @@ def _create_validation_classes(
         json_fields: list[str] = []
         for prop_name, prop in view.properties.items():
             if isinstance(prop, dm.EdgeConnection):
-                edge_by_property[prop.type.external_id] = prop_name, prop
```
Comment (nikokaoja) [inline cognite/neat/_graph/loaders/_rdf2dms.py:200]: ^ the above produce major bug as the property name is not edge type but property in view that is of type edge

Comment (doctrino) [reply cognite/neat/_graph/loaders/_rdf2dms.py:200]: Good catch.

File: tests/data/car.py (hunk 2 above)
Comment (doctrino) [reply tests/data/car.py:35]: Probably explains how the bug was introduced in the first place.

**Review by doctrino — APPROVED** [review]
See comments.

File: tests/tests_integration/test_session/test_graph_flow.py [hunk 5]
```diff
@@ -99,10 +103,53 @@ def test_dexpi_to_dms(self, cognite_client: CogniteClient, data_regression: Data
...
+        rules = neat._state.data_model.last_unverified_rule[1].rules
+        rules.metadata.created = "2024-09-19T00:00:00Z"
+        rules.metadata.updated = "2024-09-19T00:00:00Z"
+
+        neat.verify()
+
+        neat.convert("dms")
+        neat.set.data_model_id(("aml_playground", "AML", "terminology_3.0"))
+
+        if True:
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply tests/tests_integration/test_session/test_graph_flow.py:145]: Made JIRA to improve these tests as I can see your test is also not completed yet

**Pull request merged**